SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc

```diff
@@ -726,6 +724,21 @@ void Phase2TrackerDigitizerAlgorithm::pixel_inefficiency(const SubdetEfficiencie
     }
   } 
 } 
+void Phase2TrackerDigitizerAlgorithm::initializeEvent(CLHEP::HepRandomEngine* eng) {
+  if (addNoise || AddPixelInefficiency || fluctuateCharge || addThresholdSmearing) {
+    
+    flatDistribution_ = std::unique_ptr<CLHEP::RandFlat>(new CLHEP::RandFlat(*eng, 0., 1.));
+    gaussDistribution_ = std::unique_ptr<CLHEP::RandGaussQ>(new CLHEP::RandGaussQ(eng, 0., theReadoutNoise));
+  }
+  // Threshold smearing with gaussian distribution:
+  if (addThresholdSmearing) {
+    smearedThreshold_Endcap_ =  std::unique_ptr<CLHEP::RandGaussQ> (new CLHEP::RandGaussQ(eng, theThresholdInE_Endcap , theThresholdSmearing_Endcap));
+    smearedThreshold_Barrel_ = std::unique_ptr<CLHEP::RandGaussQ> (new CLHEP::RandGaussQ(eng, theThresholdInE_Barrel , theThresholdSmearing_Barrel));
```

Comment:
`std::make_unique<T>()` could be used here as well (if these are still needed, ref. @civanch's comment https://github.com/cms-sw/cmssw/pull/22836#discussion_r179105152).